src/routes/webhooks/paddle.ts

```diff
@@ -74,6 +82,39 @@ const updateUserSubscription = async ({
     );
     return false;
   }
+
+  const { gifter_id: gifterId, duration } = customData;
+  const durationTime = duration && parseInt(duration);
+  const isGift = !!durationTime && gifterId;
+  if (isGift) {
+    if (userId === gifterId) {
+      logger.error({ type: 'paddle', data }, 'User and gifter are the same');
+      return false;
+    }
```

Comment:
We don't do any protection on the frontend for this?
If not, we should :)

Reply:
Yes we should, but is better to put a check even on server, not?

Reply:
Yeah, just have to make sure we probably throw error someone goes and looks at as it should never happen hopefully.